plugins/vendor/zoom/ZoomDropFolderPlugin/batch/KZoomDropFolderEngine.php

```diff
@@ -0,0 +1,423 @@
+<?php
+/**
+ * @package plugins.ZoomDropFolder
+ */
+class KZoomDropFolderEngine extends KDropFolderFileTransferEngine
+{
+	const MAX_DATE_RANGE_DAYS = 14;
+	const ONE_DAY = 86400;
+	const MAX_PAGE_SIZE = 300;
+	const MEETINGS = 'meetings';
+	const RECORDING_FILES = 'recording_files';
+	const UUID = 'uuid';
+	const ID = 'id';
+	const TOPIC = 'topic';
+	const START_TIME = 'start_time';
+	const ACCOUNT_ID = 'account_id';
+	const HOST_ID = 'host_id';
+	const TYPE = 'type';
+	const DOWNLOAD_URL = 'download_url';
+	const RECORDING_START = 'recording_start';
+	const FILE_SIZE = 'file_size';
+	const FILE_EXTENSION = 'file_extension';
+	const RECORDING_FILE_TYPE = 'file_type';
+	const NEXT_PAGE_TOKEN = 'next_page_token';
+	const ME = 'me';
+	
+	/**
+	 * @var kZoomClient
+	 */
+	protected $zoomClient;
+	
+	protected static $lastHandledMeetingTime;
+	
+	public function watchFolder(KalturaDropFolder $dropFolder)
+	{
+		$jwtToken = isset($dropFolder->jwtToken) ? $dropFolder->jwtToken : null;

```

Comment:
Move this entire block to initZoomClient(KalturaZoomDropFolder $zoomDropFolder);
that will return the client ready for use